app/models/article.rb

```diff
@@ -522,8 +522,10 @@ def validate_tag
     return errors.add(:tag_list, "exceed the maximum of 4 tags") if tag_list.size > 4
 
     # check tags names aren't too long
+    # check that tags don't contain non alphabet characters
     tag_list.each do |tag|
       errors.add(:tag, "\"#{tag}\" is too long (maximum is 30 characters)") if tag.length > 30
```

Comment:
Rather than duplicating the tag validation logic here what if instead we used it through the model itself?
```ruby
t = Tag.new(tag)
t.valid?
t.errors.messages[:name].any?
```

Reply:
That is a great suggestion!!

Reply:
We can't actually do this because it would raise uniqueness error.